src/lib/mina_base/account.ml

```diff
@@ -762,17 +763,19 @@ let incremental_balance_between_slots ~start_slot ~end_slot ~cliff_time
     ~cliff_amount ~vesting_period ~vesting_increment ~initial_minimum_balance :
     Unsigned.UInt64.t =
   let open Unsigned in
-  let min_balance_at_start_slot =
-    min_balance_at_slot ~global_slot:start_slot ~cliff_time ~cliff_amount
-      ~vesting_period ~vesting_increment ~initial_minimum_balance
-    |> Balance.to_amount |> Amount.to_uint64
-  in
-  let min_balance_at_end_slot =
-    min_balance_at_slot ~global_slot:end_slot ~cliff_time ~cliff_amount
-      ~vesting_period ~vesting_increment ~initial_minimum_balance
-    |> Balance.to_amount |> Amount.to_uint64
-  in
-  UInt64.Infix.(min_balance_at_start_slot - min_balance_at_end_slot)
+  if Global_slot.(end_slot <= start_slot) then UInt64.zero
```

Comment:
I think I have accidentally called it with reversed arguments and observed an overflow, which is obviously bad. Hence I added a specific test, which verifies that the function behaves correctly in this case. What the correct behaviour is, is debatable. I could change it to return an `option`, although I think it makes sense to keep the types simple. In a sense, `0` is a correct result, because over `0` blocks, the incremental balance is always `0`.

If someone accidentally calls it with `end_slot` smaller than `start_slot` in the wild, it certainly means things have gone wrong somewhere else. I don't really like the idea that I have to check this function for correct result (`Ok` or `Some` or whatever) just because it could potentially be called backawrds. The caller should validate the arguments *before* calling the function, not *afterwards*. Having them pattern match on the result is punishing them for doing their job and complicating the code for the reader as well. I'd rather avoid that. I think it goes without saying that raising an exception is even worse, because it could crash a node in the middle of block production.